### PR TITLE
$fh.host - Allow lookup of URL for GUIDs, not just current app URL

### PR DESCRIPTION
**Pull request by evanshortiss**
This PR adds a new backwards compatible change to $fh.hosts to enable users to get the URL of an app besides the current one.

The reason for the added dependency `fh-instance-url` is to facilitate parsing FH_SERVICE_MAP meaning $fh.host will work during local development scenarios.

### CODE REVIEW

File: lib/host.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-var request = require('request');
+'use strict';
 
-module.exports = function host(cb) {
-  var url = 'https://' + process.env.FH_MILLICORE + '/box/srv/1.1/ide/apps/app/hosts';
-  var data = {
-    "guid": process.env.FH_INSTANCE,
-    "env": process.env.FH_ENV
-  }
+var process = require('process');
```
Comment (evanshortiss) [inline lib/host.js]: Allows us to stub process during tests

File: lib/host.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-    if (err) return cb(err);
+/**
+ * Returns the host URL for this app, or an application with the given guid
+ * @param  {String}   guid
+ * @param  {Function} cb
+ * @return {undefined}
+ */
+module.exports = function host (guid, cb) {
+  // Allow lookup of any guid on this domain
+  if (typeof guid === 'function') {
```
Comment (evanshortiss) [inline lib/host.js]: Retains backwards compatibility.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

**Comment by evanshortiss** [comment]
@wei-lee @matzew thoughts? This is something people often need, to get the URL of the MBaaS to make a manual http(s) request

File: lib/host.js [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
+    cb = guid;
+    guid = process.env.FH_INSTANCE;
+  }
 
-    return cb(err, _.getPath(body, "hosts.url"));
-  });
-}
\ No newline at end of file
+  require('fh-instance-url')({
```
Comment (evanshortiss) [inline lib/host.js]: Supports FH_SERVICE_MAP, but also does the GUID lookup if needed

**Comment by wei-lee** [comment]
I have no problem with the code change here, but not sure if we should expose this function. Also it only works if both apps are in the same domain

**Comment by wei-lee** [comment]
@evanshortiss what exactly the use case here?

**Comment by evanshortiss** [comment]
@wei-lee sometimes we need to make a request to or get the URL of an App/Service on the domain. Right now the official SDK lacks that function so this will add it. I know it has been a need for a few apps we wrote so that's why _fh-instance-url_ exists. I know @tinyexplosions and @crosbie both need it too so having it in the core SDK would be nice, but naturally you guys hold the cards.

**Pull request closed without merging**